diff --git a/src/execution/operator/persistent/physical_copy_to_file.cpp b/src/execution/operator/persistent/physical_copy_to_file.cpp
index 1c12cbd5e4..5e8a10d608 100644
--- a/src/execution/operator/persistent/physical_copy_to_file.cpp
+++ b/src/execution/operator/persistent/physical_copy_to_file.cpp
@@ -931,8 +931,15 @@ unique_ptr<const SortStrategy> PartitionedCopy::ConstructSortStrategy() const {
 	vector<BoundOrderByNode> order_bys;
 	vector<unique_ptr<BaseStatistics>> partition_stats;
 
-	return SortStrategy::Factory(context, partition_bys, order_bys, op.children[0].get().GetTypes(), partition_stats,
-	                             op.children[0].get().estimated_cardinality);
+	const vector<LogicalType> *source_types = &op.expected_types;
+	auto source_cardinality = op.estimated_cardinality;
+	if (!op.children.empty()) {
+		auto &child = op.children[0].get();
+		source_types = &child.GetTypes();
+		source_cardinality = child.estimated_cardinality;
+	}
+
+	return SortStrategy::Factory(context, partition_bys, order_bys, *source_types, partition_stats, source_cardinality);
 }
 
 void PartitionedCopy::CreateNextState() {
